src/D2L.CodeStyle.Analyzers/Immutability/ImmutableTypeInfo.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+
+namespace D2L.CodeStyle.Analyzers.Immutability {
+	/// <summary>
+	/// A representation of the immutability of some type.
+	/// </summary>
+	internal readonly struct ImmutableTypeInfo : IEquatable<ImmutableTypeInfo> {
+		public ImmutableTypeInfo(
+			ImmutableTypeKind kind,
+			ITypeSymbol type
+		) {
+			Kind = kind;
+			Type = type;
+		}
+
+		public ImmutableTypeKind Kind { get; }
+
+		public ITypeSymbol Type { get; }
+
+		public bool Equals( ImmutableTypeInfo other ) => this.Equals( other );

```

Comment:
Uh this stuff is probably not needed. Initially I had a hashset of these types but now its `Dictionary<ITypeSymbol, ImmutableTypeInfo>`.

(`ISymbol` is `IEquatable<ISymbol?>`)